Calling/ClientApp/src/core/sideEffects.ts

```diff
@@ -246,14 +264,15 @@ export const initCallAgent = (name: string, callEndedHandler: (reason: CallEndRe
         // if remote participants have changed, subscribe to the added remote participants
         addedCall.on('remoteParticipantsUpdated', (ev): void => {
           // for each of the added remote participants, subscribe to events and then just update as well in case the update has already happened
-            ev.added.forEach((addedRemoteParticipant) => {
+          ev.added.forEach((addedRemoteParticipant) => {
             subscribeToParticipant(addedRemoteParticipant, addedCall, dispatch);
-            dispatch(setParticipants([...state.calls.remoteParticipants, addedRemoteParticipant]))
+            dispatch(setParticipants([...state.calls.remoteParticipants, addedRemoteParticipant]));
           });
 
-          ev.removed.forEach((removedRemoteParticipant) => {
-            dispatch(setParticipants([...state.calls.remoteParticipants.filter(remoteParticipant => { return remoteParticipant !== removedRemoteParticipant }) ]))
-          });
+          // We don't use the actual value we are just going to reset the remoteParticipants based on the call
+          if (ev.removed.length > 0) {
+            dispatch(setParticipants([...addedCall.remoteParticipants.values()]));
```

Comment:
This is not directly related to the PR. 
It fixes a bug when a remote user disconnect the call, all remote participants get reset, it was fixed in the old repo but looks like it wasn't ported over [link](https://skype.visualstudio.com/SCC/_git/internal_service_spool-e2e?path=%2FCalling%2FClientApp%2Fsrc%2Fcore%2FsideEffects.ts&version=GBcalling-ui-sample&line=169&lineEnd=173&lineStartColumn=13&lineEndColumn=14&lineStyle=plain&_a=contents)